.github/workflows/test.yml

```diff
@@ -12,16 +12,31 @@ jobs:
       uses: actions/checkout@v2
       with:
         fetch-depth: 0
-    - name: setup python
-      uses: actions/setup-python@v4
+    - name: Cache conda
+      uses: actions/cache@v2
+      env:
+        # Increase this value to reset cache if etc/example-environment.yml has not changed
+        CACHE_NUMBER: 0
       with:
-        cache: pip
-        cache-dependency-path: pyproject.toml
-        python-version: "${{ matrix.python-version}}"
-    - name: upgrade dependencies
-      run: |-
-        python -m pip install --upgrade pip hatch
-        pip install -e.[test]
+        path: ~/conda_pkgs_dir
+        key:
+          ${{ runner.os }}-conda-${{ env.CACHE_NUMBER }}-${{
+          hashFiles('test-environment.yml') }}
+    - uses: conda-incubator/setup-miniconda@v2
+      with:
+        activate-environment: test-nbconvert-a11y

```

Comment:
```yaml
with:
        channel-priority: strict
        environment-file: test-environment.yml
        miniforge-variant: Mambaforge
        use-mamba: true
```

Then you need the shell defaults, either at the job or pipeline level, so that the setup scripts are respected:

```yaml
defaults:
  run:
    shell: bash -el {0}
```

Reply:
thanks. i felt like i was missing that shell bit. i changed the config like you suggested.